### PR TITLE
added assessments api

### PR DESCRIPTION
**Pull request by somnath77**

### CODE REVIEW

File: app/controllers/api/v1/admin/assessments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        def fetch_drive_data
+          @drive = Drive.find(params[:assessment_id])
+        end
+
+        def drive_params
+          params.permit(:name, :description, :start_time, :end_time, drives_problems_attributes: %i[id problem_id _destroy])
+        end
+
+        def authenticate_token
+          render_error(message: I18n.t('devise.failure.unauthenticated')) unless request.headers["HTTP_AUTH_TOKEN"] == Organization.first.auth_token
```
Comment (anilmaurya) [inline app/controllers/api/v1/admin/assessments_controller.rb:47]: Add TODO 

Organization should be loaded from drive

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+  let(:problem) { create(:problem, created_by_id: user.id, updated_by_id: user.id) }
+
+  describe 'GET#index' do
+    before do
+      create(:drive, created_by_id: user.id, updated_by_id: user.id, organization: organization, is_assessment: true)
+      create(:drive, created_by_id: user.id, updated_by_id: user.id, organization: organization)
+      create(:drive, created_by_id: user.id, updated_by_id: user.id, organization: organization, is_assessment: true)
+    end
+    context 'when user is logged in' do
```
Comment (anilmaurya) [inline spec/controllers/api/v1/admin/assessments_controller_spec.rb:19]: statement is no correct, correct statement will be
```
when auth token is valid
```

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        request.headers.merge!(HTTP_AUTH_TOKEN: 'bearer_token')
+        get :index
+
+        data = json
+
+        expect(data['data']['assessments'].count).to eq(2)
+        expect(response).to have_http_status(:ok)
+      end
+    end
+    context 'When user is not logged in' do
```
Comment (anilmaurya) [inline spec/controllers/api/v1/admin/assessments_controller_spec.rb:30]: Statement not correct

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          expect(data.first).to eq(I18n.t('drive_end_time.invalid'))
+        end
+      end
+    end
+
+    context 'When user is not logged in' do
+      it ' ask for login ' do
+        post :create, params: { assessment_id: drive.id, email: Faker::Internet.email }
+
+        expect(response.body).to eq('You need to sign in or sign up before continuing.')
```
Comment (anilmaurya) [inline spec/controllers/api/v1/admin/assessments_controller_spec.rb:72]: Auth token is not valid

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          request.headers.merge!(HTTP_AUTH_TOKEN: 'bearer_token')
+          post :create, params: { assessment_id: drive.id, email: Faker::Internet.email, start_time: DateTime.current, end_time: DateTime.current - 30.minutes }
+          data = json
+
+          expect(data.first).to eq(I18n.t('drive_end_time.invalid'))
+        end
+      end
+    end
+
+    context 'When user is not logged in' do
```
Comment (anilmaurya) [inline spec/controllers/api/v1/admin/assessments_controller_spec.rb:68]: statement not correct

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb (hunk 2 above)
Comment (somnath77) [reply spec/controllers/api/v1/admin/assessments_controller_spec.rb:19]: done

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb (hunk 3 above)
Comment (somnath77) [reply spec/controllers/api/v1/admin/assessments_controller_spec.rb:30]: changed

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb (hunk 4 above)
Comment (somnath77) [reply spec/controllers/api/v1/admin/assessments_controller_spec.rb:72]: changed

File: spec/controllers/api/v1/admin/assessments_controller_spec.rb (hunk 5 above)
Comment (somnath77) [reply spec/controllers/api/v1/admin/assessments_controller_spec.rb:68]: changed

**Pull request merged**